Added Navbar component

cheina97 opened the pull request:
Components added:

- Navbar (common)

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

QcFe commented:
I'm not really convinced with this approach actually. To keep things decoupled I'd avoid working with history here at all. I'd rather pass a function from the parent with the setter of the current view. Then such parent will be responsible of handling such change. 
(Also I'd make small custom components and/or adding a loop to avoid that heavy code replication)
Any thought @GabriFila?

cheina97 commented:
> I'm not really convinced with this approach actually. To keep things decoupled I'd avoid working with history here at all. I'd rather pass a function from the parent with the setter of the current view. Then such parent will be responsible of handling such change.
> (Also I'd make small custom components and/or adding a loop to avoid that heavy code replication)
> Any thought @GabriFila?

Ok i'll find a way to avoid code duplication

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

QcFe commented on frontend/src/components/common/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,74 @@
+  const [show, setShow] = useState(false);
+
+  const buttons = [
+    { path: '/', name: 'main' },
+    { path: '/active', name: 'active' },
+    { path: '/drive', name: 'drive' },
+    { path: '/account', name: 'account' },
```
I'd actually put the display label here, so with the uppercase already, then pass and use setActiveRoute from the parent instead of setActiveLabel

cheina97 replied:
But how can i change the passed state clicking the buttons without a setState function? Maybe i'm misunderstanding

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
Reopen PR

cheina97 commented:
/rebase

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

giorio94 commented on frontend/src/components/common/Logo/Logo.tsx:
```diff
@@ -0,0 +1,23 @@
+  const { width } = props;
+  return (
+    <img
+      src={isDarkTheme ? logoDarkTheme : logoLightTheme}
```
Since the difference between the two logos seems to be only related to color, could it be changed dynamically through CSS instead of having two versions?

cheina97 replied:
I tried to change it using CSS, but without success. Maybe would be better to discuss about it during the meeting. According to the react documentation https://it.reactjs.org/docs/context.html#when-to-use-context is a good idea to use a context to choose the used theme. Maybe i could be try to hardcode the svg in the logo component to avoid file duplication (i'm talking about logo.svg and logoLightTheme.svg)

giorio94 commented on frontend/src/components/common/Logo/Logo.tsx:
```diff
@@ -0,0 +1,23 @@
+    <img
+      src={isDarkTheme ? logoDarkTheme : logoLightTheme}
+      width={width}
+      alt="Crownlabs Logo"
```
```suggestion
      alt="CrownLabs Logo"
```

giorio94 commented on frontend/src/Decorators.tsx:
```diff
@@ -1,8 +1,12 @@
+/* eslint-disable react/no-multi-comp */
```
Should add a comment why this is needed.

cheina97 replied:
I just added a comment. After this PR, as agreed with gabriele, we will do a "refactoring" of the decorators by dividing them into several files. This file will be removed.

giorio94 reviewed (approved):
From a user perspective, looks mostly good to me.

Was just wondering whether to put the theme switcher inside the hamburger menu in the mobile view. WDYT?

cheina97 commented:
> From a user perspective, looks mostly good to me.
> 
> Was just wondering whether to put the theme switcher inside the hamburger menu in the mobile view. WDYT?

I added an alternative behaviour for the responsive mode. I modified and pushed app.tsx just to show you the correct behaviour of the component, that doesn't work correctly in the storybook. Before merging this PR i will restore the old app.tsx

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

giorio94 commented:
I personally like the current implementation. Do you think an intermediate status with only main, active and the hamburger for the others could be useful? I was mainly thinking about a split screen scenario, with the navbar which now looks empty.

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/deploy-staging

cheina97 commented:
/hold

cheina97 commented:
/rebase

cheina97 commented:
/deploy-staging

cheina97 commented:
/deploy-staging

cheina97 commented:
/deploy-staging

cheina97 commented:
/deploy-staging

cheina97 commented:
/deploy-staging

cheina97 commented:
deploy-staging

cheina97 commented:
/deploy-staging

cheina97 commented:
/deploy-staging

cheina97 commented:
/deploy-staging

cheina97 commented:
/deploy-staging

GabriFila commented:
/rebase

GabriFila commented:
/rebase

GabriFila commented:
/merge
